radixdlt-java/src/main/java/com/radixdlt/client/core/atoms/Hashable.java

```diff
@@ -0,0 +1,23 @@
+package com.radixdlt.client.core.atoms;
+
+import org.radix.common.ID.EUID;
+
+/**
+ * Representing any type that can be *hashed* into a {@link RadixHash}
+ * with a utility method to obtain an HID from that hash.
+ */
+public interface Hashable {
```

Comment:
This interface isn't correct.

First issue is that it isn't really being used as an interface but rather for inheritance but now in a worse way because the interface suggests that `hid()` can be modified.

Secondly, the hash of an object shouldn't be calculated by the class itself as we have a very specific mechanism with DSON on how to calculate these hashes and Hash calculation and classes should be decoupled in the long run so these methods shouldn't be supported by an interface.

Reply:
Fair point. Will remove.